lib/auth/init.go

```diff
@@ -106,6 +106,14 @@ type InitConfig struct {
 
 	// U2F defines U2F application ID and any facets passed in from a configuration file.
 	U2F services.UniversalSecondFactor
+
+	// TrustedClusterBackend is a service to do CRUD operations on a TrustedCluster resource.
+	TrustedClusterBackend services.TrustedClusterBackend
```

Comment:
I don't think you need a backend per resource, you can use `Presence` service to host `TrustedCluster`